src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -267,25 +267,44 @@ public void validateInputFiles() throws Exception {
     }
 
     private static void validateInputDirectory(Path checkDir) throws IOException {
-        final Pattern pattern = Pattern.compile("^SuppressionXpathRegression(.+)\\.java$");
+        final Pattern patternTemp = Pattern.compile("^SuppressionXpathRegression(.+)\\.java$");
+        final Pattern pattern = Pattern.compile("^InputXpath(.+)\\.java$");
+
         final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(checkDir.toFile().getName());
 
         try (DirectoryStream<Path> inputPaths = Files.newDirectoryStream(checkDir)) {
             for (Path inputPath : inputPaths) {
                 final String filename = inputPath.toFile().getName();
-                if (filename.endsWith("java")) {
-                    final Matcher matcher = pattern.matcher(filename);
+                if (filename.contains("SuppressionXpathRegression") && filename.endsWith("java")) {
```

Comment:
It was actually observed here..
At first, I roughly used : 
```
if (filename.endsWith("java") {
    if (filename.contains("SuppressionXpathRegression")) {
       // code block
    }
}

```
An error was reported saying the ifs could be combined, which is obvious

Reply:
The line you are commenting on is a collapsed nested if, which you said in the first post wasn't acceptable. :)
https://jsparrow.github.io/rules/collapse-if-statements.html#simple-collapse

Reply:
Yep, it is good that we check where ifs can be combined. ;)
However, I remember that day vigorously browsing through the codebase for the class/method which checks this.
Still am infact, but I somehow don't seem to find it....